### ISSUE TITLE
input_filename_base is inconsistent

### ISSUE THREAD

**Issue by nordurljosahvida** [OrcaSlicer/OrcaSlicer#1945]
### Bambu Studio Version

1.6.6

### Where is the application from?

Bambu Lab github releases

### OS version

win11

### Printer

x1 carbon

### How to reproduce

- open a new project
- select any bambu printer [i'm using x1 carbon]
- import any geometry or test cube
- save the project with a custom name
- press "slice"
- press "export gcode file"

or with any custom printer:

- open a new project
- select a non-bambu printer [any custom one you created, i'm using one based on the generic klipper printer that can be added on first setup of orca]
- add any geometry or test cube
- save the project
- press "slice"
- export gcode or press the "print" button

### Actual results

the name of the file will be the name of the object being printed, not of the actual saved project name

### Expected results

The exported gcode name includes {input_filename_base} as per output options setting. However, this variable should the project name, not the imported model geometry file's [the stl as an example]. In prusaslicer, it's exactly like this. Even in bambu and orca, when sending a file to a bambu printer, the name is of the projects, this is only an issue in how input_filename_base is processed.

Of course, if a project is saved, then the exported gcode - whether exported locally or to sd card or sent over directly to printer via octoprint / klipper - should match that name. I think this problem was introduced in bambu studio, but is very easily fixable.

Thanks!

### Project file & Debug log uploads

no file as this should be tested from scratch with any geometry

### Checklist of files to include

- [ ] Log file
- [ ] Project file

**Comment by SylvainKlein** [comment]
Having the exact same issue here. Would love a fix for that ! :)

**Comment by Matt2916** [comment]
Still present in Orca 1.8.1 - would love a fix for this!

**Comment by UnseenStyle** [comment]
Would love the ability to label bambu print filenames to the plate name instead of it being just the project name.

**Comment by vale075** [comment]
> Would love the ability to label bambu print filenames to the plate name instead of it being just the project name.

It is already possible by using: plate_name

**Comment by jbjhjm** [comment]
A fix for this would be much appreciated. Currently I need to manually change the file name for each export.

**Comment by jbjhjm** [comment]
I took a moment to check the source code. 
Interestingly, the implementation has inconsistent concepts in all code bases, be it orca, prusa, superslicer or bambu.

https://github.com/SoftFever/OrcaSlicer/blob/main/src/libslic3r/PrintBase.cpp#L58
contains two functions defining different values to input_filename / input_filename_base.

In PrintBase::update_object_placeholders these two are set to the INPUT MODEL filename.
In PrintBase::output_filename these two are set to the PROJECT file name.

The latter one will only everwrite the config values though if a non-empty filename_base argument is being passed into the function. I'm currently assuming that for some reason this is not happening for Orca anymore.

It seems that the output_filename function is regularly called by Print::output_filename.
That's where I'm stuck right now, I cannot seem to figure out what's calling that function and passing a valid/invalid file name.

The existing implementation seems questionworthy, when "fixing the override" I guess other users may be irritated.

**Comment by jbjhjm** [comment]
chu bot, nothing stale here!

**Comment by Jeppedy** [comment]
Yup, just trying out OrcaSlicer, and that's the first thing I noticed..  My filename has "body1" as the name, and not the ACTUAL project file name.  I could use plate_name, but I still have no reliable way to introduce the "project" name to the filename.

**Comment by ShryuKt** [comment]
I'm looking for the same thing

**Comment by poulanker** [comment]
I also noticed, that no matter which AMS spool number you select for slicing, the filament type inserted in the gcode filename always follows whatever filament type is selected for spool #1 .

**Comment by Jeppedy** [comment]
For those looking for a quick hack/fix, I set my model name in the Project tab and use the following Filename template:
{if model_name!=""}{model_name}_{endif}{if plate_name!=""}{plate_name}_{endif}{if filament_type[0]!=""}{filament_vendor[0]}{filament_type[0]}{else}NoFilamentType{endif}_{nozzle_diameter[0]}.gcode

**Comment by blalor** [comment]
Not stale, still exists. The current behavior is really weird, especially coming from PrusaSlicer.

**Comment by golovashevich** [comment]
Still exists

**Comment by fazadaoc** [comment]
> For those looking for a quick hack/fix, I set my model name in the Project tab and use the following Filename template: {if model_name!=""}{model_name}_{endif}{if plate_name!=""}{plate_name}_{endif}{if filament_type[0]!=""}{filament_vendor[0]}{filament_type[0]}{else}NoFilamentType{endif}_{nozzle_diameter[0]}.gcode

thanks for this and this [#987](https://github.com/SoftFever/OrcaSlicer/issues/987#issuecomment-2701761387)

this is my code `{print_time}_{if model_name!=""}{model_name}{else}{input_filename_base}{endif}{if plate_name!="Untitled"}_{plate_name}{endif}.gcode`

for some reason thought instead of using the file name (as in the name of the file itself as shown in top bar (the .stl/.3mf) **not** the "model name" set in the project tab), i am getting the name of the first object imported on the plate as reported here [#2752](https://github.com/SoftFever/OrcaSlicer/issues/2752#issue-1995301609)

Anyone else able to check on this behaviour? thanks

**Comment by Jeppedy** [comment]
Still hoping for a formal resolution

*"What good is it for a man to gain the whole world, yet forfeit his soul?"*
- *Mark 8:36
<http://christianbookshelf.org/spurgeon/spurgeons_sermons_volume_2_1856/profit_and_loss.htm>*

**Comment by Otann** [comment]
Would also love to have project name as print job name

**Comment by greencheetah100** [comment]
I still have this issue in 2.3.1

{input_filename} and {input_filename_base} both return Object_1, the default first object name. Same result with custom printer and ender 3 preset.
